connections/github.py

```diff
@@ -101,8 +101,14 @@ def get_instance():
     global _gh, _token
     # TODO Possibly renew token already if expiry date is soon, not only
     #      after it has expired.
-    if not _gh or (_token and datetime.datetime.utcnow() > _token.expires_at):
-        _gh = connect()
+
+    # Check if PyGithub version is < 1.56
+    if hasattr(github, 'GithubRetry') is False:
+        if not _gh or (_token and datetime.utcnow() > _token.expires_at):
+            _gh = connect()
+    elif hasattr(github, 'GithubRetry') is True:
+        if not _gh or (_token and datetime.now(timezone.utc) > _token.expires_at):
+            _gh = connect()
```

Comment:
`if ... is False` is kind of silly, and positive logic is easier to follow, and there's a lot of copy-pasting, so I would change this to:

```python
    # PyGithub 2.x expects a datetime object with timezone info,
    # PyGithub 1.x requires one without timezone info;
    # see also https://github.com/PyGithub/PyGithub/pull/2565
    if hasattr(github, 'GithubRetry'):
        # PyGithbub 2.x
        time_now = datetime.now(timezone.utc)
    else:
        # PyGithbub 1.x
        time_now = datetime.utcnow()

    if not _gh or (_token and time_now > _token.expires_at):
        _gh = connect()